features/FEATURE_BLE/targets/TARGET_CORDIO/stack_adaptation/hci_tr.c

```diff
@@ -95,12 +95,13 @@ void hciTrSendCmd(uint8_t *pData)
   /* dump event for protocol analysis */
   HCI_PDUMP_CMD(len, pData);
 
-  /* transmit ACL header and data */
-  if (hciDrvWrite(HCI_CMD_TYPE, len, pData) == len)
+  /* transmit ACL header and data (releases the ownership of the WSF buffer containing pData) */
+  if (hciDrvWrite(HCI_CMD_TYPE, len, pData) != len)
   {
-    /* free buffer */
-    WsfMsgFree(pData);
+    /* transport failure */
```

Comment:
Nope, that's going to break the interface on targets that don't use zero copy (All of them except the new Cordio LL). This can be solved with a compile time configuration; if the target support zero copy then we do not release the data, otherwise we release them. No zero copy should be the default mode.